### DISCUSSION TITLE
[Question] How Lengthscale prior affects the model training process?

### DISCUSSION THREAD

**Discussion by mohammad-saber** [cornellius-gp/gpytorch#1782] [Q&A] [answered]
Hi, 
Thank you for sharing your great work. 

I have a misunderstanding about Kernel `lengthscale_prior`.  I appreciate it if you could guide me.

When I set Kernel `lengthscale_prior`, is it only used to set initial value of kernel lengthscale? or is it also effective in model training process?

My understanding is that kernel lengthscale is obtained by optimizing Marginal Likelihood through training process. I would like to ask if `lengthscale_prior`is also used in training process and how it affects the value of kernel lengthscale in model trianing.

Thank you for your time to read my question.

**Comment by wjmaddox** [comment, marked as answer]
The log likelihood of the prior wrt the lengthscale is added into the log marginal likelihood during the training process.

Note that to set the lengthscale to a sample from the prior, you'll still have to draw a sample from the prior itself `covar.lengthscale = covar.lengthscale_prior.rsample()`.

**Reply by patel-zeel** [reply]
[Exact position in the source code](https://github.com/cornellius-gp/gpytorch/blob/fc2053b0fc00517880fbc11adc7f5802242eec6a/gpytorch/mlls/exact_marginal_log_likelihood.py#L41)

**Reply by ghost** [reply]
is this then the type II maximum likelihood ?

**Reply by wjmaddox** [reply]
In general, yes. Unless you're doing fully Bayesian inference which is also supported.
